packages/repository/src/repositories/migrateable.repository.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Options} from '../common-types';
+import {Model} from '../model';
+import {Repository} from './repository';
+
+export interface SchemaMigrationOptions extends Options {
+  /**
+   * When set to true, schema migration will drop existing tables and recreate
+   * them from scratch, removing any existing data along the way.
+   */
+  rebuild?: boolean;
```

Comment:
We should have an option to dry run - returning a bunch of SQL statements.

Reply:
+1

Unfortunately, our current juggler API does not provide dry-run feature, therefore I don't have any easy means for enabling dry run option. Let's leave it out of scope of this pull request.

In the future, I would like us to implement a mode where we create a SQL file containing the migration script, so that the user can customize it. See https://github.com/strongloop/loopback-next/issues/487.